src/OrchardCore.Modules/OrchardCore.ContentFields/Services/DefaultContentPickerResultProvider.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using YesSql;
+using YesSql.Services;
+
+namespace OrchardCore.ContentFields.Services
+{
+    public class DefaultContentPickerResultProvider : IContentPickerResultProvider
+    {
+        private readonly IContentManager _contentManager;
+        private readonly ISession _session;
+
+        public DefaultContentPickerResultProvider(IContentManager contentManager, ISession session)
+        {
+            _contentManager = contentManager;
+            _session = session;
+        }
+
+        public string Name => "Default";
+
+        public async Task<IEnumerable<ContentPickerResult>> Search(ContentPickerSearchContext searchContext)
+        {
+            var contentItems = await _session.Query<ContentItem, ContentItemIndex>()
+                .With<ContentItemIndex>(x => x.ContentType.IsIn(searchContext.ContentTypes) && x.Published)
+                .ListAsync();
+
+            var results = new List<ContentPickerResult>();
+
+            foreach (var contentItem in contentItems)
+            {
+                var contentItemMetadata = await _contentManager.PopulateAspectAsync<ContentItemMetadata>(contentItem);
+                results.Add(new ContentPickerResult
+                {
+                    ContentItemId = contentItem.ContentItemId,
+                    DisplayText = contentItemMetadata.DisplayText
```

Comment:
Makes me think we could provide a YesSql index that would contain the DisplayText and the BodyText with full text search options, if it's available in all databases. Or at least the display text without full text. Super easy to do and would solve many search scenarios from the admin. Lucene is always tricky to manage. And this should work with any entity so we could have other things in this list, like Users, ... I will look into it, don't bother with it.

Reply:
Yeah that would be great.